index.js

```diff
@@ -1,47 +1,246 @@
-var swarmDefaults = require('dat-swarm-defaults')
-var disc = require('discovery-swarm')
-var xtend = require('xtend')
-
-module.exports = HyperdriveSwarm
-
-function HyperdriveSwarm (archive, opts) {
-  if (!(this instanceof HyperdriveSwarm)) return new HyperdriveSwarm(archive, opts)
-  if (!opts) opts = {}
-
-  var self = this
-  this.archive = archive
-  this.uploading = !(opts.upload === false)
-  this.downloading = !(opts.download === false)
-  this.live = !(opts.live === false)
-
-  var isHyperdbInstance = !!(archive.get && archive.put && archive.replicate && archive.authorize)
-
-  if (isHyperdbInstance && !archive.local) {
-    throw new Error('hyperdiscovery swarm must be created after the local hyperdb instance is ready!')
-  }
-
-  // Discovery Swarm Options
-  opts = xtend({
-    port: 3282,
-    id: isHyperdbInstance ? archive.local.id.toString('hex') : archive.id,
-    hash: false,
-    stream: function (peer) {
-      return archive.replicate(xtend({
-        live: self.live,
-        upload: self.uploading,
-        download: self.downloading
-      }, isHyperdbInstance ? {
-        userData: archive.local.key
-      } : {}))
-    }
-  }, opts)
-
-  this.swarm = disc(swarmDefaults(opts))
-  this.swarm.once('error', function () {
-    self.swarm.listen(0)
-  })
-
-  this.swarm.listen(opts.port)
-  this.swarm.join(this.archive.discoveryKey)
-  return this.swarm
+const crypto = require('crypto')
+const EventEmitter = require('events')
+
+const datEncoding = require('dat-encoding')
+const hypercoreProtocol = require('hypercore-protocol')
+const discoverySwarm = require('discovery-swarm')
+const swarmDefaults = require('dat-swarm-defaults')
+const mutexify = require('mutexify')
+const getPort = require('get-port')
+
+const debug = require('debug')('hyperdiscovery')
+
+module.exports = (...args) => new Hyperdiscovery(...args)
+
+const DEFAULT_PORT = 3282
+const ALTERNATE_PORTS = [3000, 3002, 3004, 2001, 2003, 2005]
```

Comment:
We can start moving to using a range (or set) of alternate ports. Any suggestions?

Reply:
DEFAULT_PORT and ALTERNATE_PORTS should be the same thing: PORTS

`DEFAULT_PORTS = [3282, 3000, ....]`